cli/Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -74,92 +88,175 @@ public struct ServerAuthenticationController: ServerAuthenticationControlling {
         #endif
     }
 
+    public func refreshToken(serverURL: URL, inSubprocess: Bool) async throws {
+        try await authenticationTokenRefreshingIfNeeded(
+            serverURL: serverURL,
+            forceRefresh: true,
+            inSubprocess: inSubprocess
+        )
+    }
+
     public func refreshToken(serverURL: URL) async throws {
         try await authenticationTokenRefreshingIfNeeded(
-            serverURL: serverURL, forceRefresh: true
+            serverURL: serverURL,
+            forceRefresh: true,
+            inSubprocess: true
         )
     }
 
     @discardableResult private func authenticationTokenRefreshingIfNeeded(
         serverURL: URL,
-        forceRefresh: Bool
+        forceRefresh: Bool,
+        inSubprocess: Bool
     ) async throws
         -> AuthenticationToken?
     {
         #if canImport(TuistSupport)
             Logger.current.debug("Refreshing authentication token for \(serverURL) if needed")
         #endif
-        return
-            try await CachedValueStore.current
-                .getValue(key: "token_\(serverURL.absoluteString)") {
-                    () -> (value: AuthenticationToken, expiresAt: Date?)? in
-                    guard let token = try await fetchTokenFromStore(serverURL: serverURL) else {
-                        return nil
-                    }
+        switch (try await tokenStatus(serverURL: serverURL, forceRefresh: forceRefresh), inSubprocess) {
+        case let (.valid(token), _):
+            return token
+        case (.absent, _), (_, false):
+            return try await executeRefresh(serverURL: serverURL, forceRefresh: forceRefresh)?.value
+        case (.expired, true):
+            let lockfilePath = lockFilePath(serverURL: serverURL)
+
+            if !(try await fileSystem.exists(lockfilePath)) {
+                try await fileSystem.touch(lockfilePath)
+
+                let process = Process()
+                process.environment = ProcessInfo.processInfo.environment
+                process.launchPath = Environment.current.currentExecutablePath()?.pathString ?? ""
+                process.arguments = [
```

Comment:
Since we removed `CachedValueStore.current` ... won't the background process be run multiple times from multiple threads? I'd be for keeping it and trigger a background process just once.

I understand there's a lock in the background process of `refresh-token`, but I'd rather be conservative with spawning the number of background processes. Data races will always be easier to manage in a single process.